numpy/core/src/multiarray/mapping.c

```diff
@@ -2479,13 +2480,18 @@ PyArray_MapIterCheckIndices(PyArrayMapIterObject *mit)
     int i;
     NPY_BEGIN_THREADS_DEF;
 
-    if (mit->size == 0) {
-        /* All indices got broadcast away, do *not* check as it always was */
+    intp_type = PyArray_DescrFromType(NPY_INTP);
+
+    if (mit->size == 0 && NpyIter_GetIterSize(mit->outer) == 0) {
```

Comment:
Does the latter imply the former?

Reply:
Yeah, it does can just remove it, might micro-optimize, but at this point probably worthless.
```suggestion
    if (NpyIter_GetIterSize(mit->outer) == 0) {
```

Reply:
Namely, would this work too?
```suggestion
    if (NpyIter_GetIterSize(mit->outer) == 0) {
```